Carnap/src/Carnap/Languages/PurePropositional/Logic/Ripley.hs

```diff
@@ -55,9 +57,13 @@ instance Inference RipleyLNJ PurePropLexicon (Form Bool) where
         ruleOf DisjElim   = proofByCasesVariations !! 0
         ruleOf As         = axiom        
 
-        indirectInference x
-           | x `elem` [CondIntro, DisjElim] = Just PolyProof
-           | otherwise = Nothing
+        globalRestriction (Left ded) n CondIntro = Just (dischargeConstraint n ded (view lhs $ conclusionOf CondElim))
+        globalRestriction (Left ded) n DisjElim = Just (dischargeConstraint n ded (view lhs $ conclusionOf DisjElim))
```

Comment:
The global restriction here is an extra check to make sure that you are removing the right dependencies when you discharge assumptions using `CondIntro` and `DisjElim`. Without these, Carnap only makes sure that the right number of dependencies have been removed.

Global restrictions are constraints on what works that operate at the level of `Deductions` or `DeductionTrees` rather than `ProofTrees` (e.g. a constraint on ∃-elim which requires that the constant you introduce doesn't occur on any prior line). The lemmon thing needs to be a globalRestriction because we throw away the Lemmon style dependency information when we extract the ProofTree from the Deduction.